### ISSUE TITLE
[Bug]: The heading value comes back as -1 on a physical iOS device

### ISSUE THREAD

**Issue by justintoth** [Baseflow/flutter-geolocator#1281]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-geolocator/blob/main/geolocator/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [ ] Android
- [X] iOS
- [ ] Linux
- [ ] macOS
- [ ] Web
- [ ] Windows

### Steps to reproduce

The lastPosition.heading value is coming back as -1.0 on a physical iOS device (iPhone XR). I understand from the Apple documentation that a -1 value indicates no heading found, however this isn't an issue with the iPhone. If I download another app that shows the heading (SeaNav), it correctly shows and updates it as I move the iPhone around. Also, the flutter_compass plugin correctly retrieves the value, so this issue is specific to the geolocator plugin.

### Expected results

That the correct heading value would be returned.

### Actual results

A heading value of -1 is returned.

### Code sample

<details><summary>Code sample</summary>

```dart
final lastPosition = await Geolocator.getCurrentPosition(desiredAccuracy: LocationAccuracy.best, timeLimit: const Duration(seconds: 1));
print('heading: ${lastPosition.heading}');
```

</details>

### Version

9.0.2

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[!] Flutter (Channel unknown, 3.7.12, on macOS 13.4.1 22F82 darwin-arm64 (Rosetta), locale en-US)
    ! Flutter version 3.7.12 on channel unknown at /Library/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions at
      https://flutter.dev/docs/get-started/install.
    ! Unknown upstream repository.
      Reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
    • Framework revision 4d9e56e694 (10 weeks ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to
      perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/justintoth/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • ANDROID_HOME = /Users/justintoth/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/justintoth/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.79.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (3 available)
    • Justin’s iPhone XR (mobile) • 00008020-00191D681E92002E • ios            • iOS 16.5.1 20F75
    • macOS (desktop)             • macos                     • darwin-arm64   • macOS 13.4.1 22F82 darwin-arm64 (Rosetta)
    • Chrome (web)                • chrome                    • web-javascript • Google Chrome 114.0.5735.133
    ! Error: Justin’s iPhone XR is busy: Fetching debug symbols for Justin’s iPhone XR. Xcode will continue when Justin’s
      iPhone XR is finished. (code -10)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by rmortes** [comment]
Con confirm this is an issue

**Comment by TimHoogstrate** [comment]
Dear @jpnurmi,

I've tested it on a real iPhone 13 iOs 17.1 and it seems to work as expected. However, heading might be a bit unclear to understand. But in iOS case `heading` means `course`. It should be available when the phone is traveling outside. You can try to test this by walking outside with the example app and then you should see a `heading`, basically it is a [course](https://developer.apple.com/documentation/corelocation/cllocation/1423832-course?language=objc).

**Comment by ICP-Theo** [comment]
I had the same issue, and changed the code in  geolocator_apple-2.3.2.
File: ios/Classes/Utils/ LocationMapper.m -> line 62
The availability test probobly is causing the problem, if I insert a copy of line 69 before line 62 then heading (course) works just fine. So please check this issue so it van be solved in the next release.

  I have seen problems with availabe attrribute with other users, see: https://stackoverflow.com/questions/72751243/available-attribute-incorrectly-constrained-on-macos-version

**Comment by TimHoogstrate** [comment]
Dear @ICP-Theo,

Thanks for your addition. I noticed a difference between the availability check regarding macOS and OSX. Can you tell me on what device/os version you are testing your app?

Kind regards,

**Comment by TimHoogstrate** [comment]
Adding to my original story; we found out that the difference between the flutter_compass plugin and ours is that the one user [CLHeading] (https://developer.apple.com/documentation/corelocation/clheading) and ours uses [CLLocation] (https://developer.apple.com/documentation/corelocation/cllocation). So, the flutter compass uses the compass (magnetic heading) and not the course (direction of travel). However, we have decided to make a little change and pass the CLLocation [courseAccuracy] (CLLocationDirectionAccuracy](https://developer.apple.com/documentation/corelocation/cllocation/3524338-courseaccuracy) to the developer even if the number is -1 (not available), so that the developer can decide what to do with it. 

Kind regards,
